cmd/kubeapps-apis/server/server.go

```diff
@@ -95,3 +96,17 @@ type gwHandlerArgs struct {
 	addr        string
 	dialOptions []grpc.DialOption
 }
+
+// Create a gateway mux that does not emit unpopulated fields.
+func gatewayMux() *runtime.ServeMux {
```

Comment:
Ah, nice, so here is where we should put something like [1] if we wanna support real HTTP status code when serving the rest-ish API

[1] https://grpc-ecosystem.github.io/grpc-gateway/docs/mapping/customizing_your_gateway/#customizing-routing-errors

Reply:
Perhaps, if we need to customise them, but the default conversions done by grpc-gateway should be what we need though, right? https://github.com/grpc-ecosystem/grpc-gateway/blob/master/runtime/errors.go#L36

I think the doc you linked to is just for being able to customise the *routing* errors, the default implementation for which is at https://github.com/grpc-ecosystem/grpc-gateway/blob/master/runtime/errors.go#L165 ?

Reply:
Ahh, great, so we already have default conversions even for the routing. Good to know!  When I saw the response bodies (`{ "code": 5, "message": "Not Found" }`) I initially thought they weren't being converted, but, yep, having a look at the whole HTTP response, the status codes are correct.
Thank you for letting me know!